File: samples/maui/KsDialogs.Sample.Maui/DeclarativeDialogViewModel.cs
namespace KsDialogs.Sample.Maui;

/// <summary>
/// Declarative Dialog の ViewModel。
/// </summary>
/// <remarks>
/// 結果型を書かずに済む顔で宣言しているため、結果は真偽値になる (core/ADR-0012)。
/// この型そのものがレジストリの登録キーになる。
/// </remarks>
/// <param name="message">ダイアログに表示するメッセージ。</param>
public sealed class DeclarativeDialogViewModel(string message) : IDialogViewModel
{
    /// <summary>ダイアログに表示するメッセージ。</summary>
    public string Message { get; } = message;
}

File: samples/kmp/shared/src/commonMain/kotlin/jp/kamusoft/ksdialogs/samples/kmp/DeclarativeDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.kmp

import jp.kamusoft.ksdialogs.kmp.DialogViewModel

/**
 * Declarative Dialog の ViewModel。
 *
 * 結果型は [Boolean] で、完了操作は `true` を報告する。
 * 共有コードで定義したこのクラスがそのまま各 OS のレジストリの登録キーになる。
 *
 * @property message ダイアログに表示するメッセージ
 */
class DeclarativeDialogViewModel(val message: String) : DialogViewModel<Boolean>
